### PR TITLE
[CleanUp] Cleans Garnet Spec Implementation

### PR DESCRIPTION
**Pull request by eliasjpr**
Controller Specs
- Changes how Controller tests are performed it now requires less
boilerplate code

System Tests

- Removes unecessary modules and simplifies code a lot.
- No need to specify Capabilities
- Class change from GarnetSpec::System::Test to GarnetSpec::SystemTest

```crystal
require "./spec_helper"
include GarnetSpec::RequestHelper

describe "Request Specs" do
  describe "#get" do
    it "should return a HTTP::Client::Response" do
      get("/posts").should be_a HTTP::Client::Response
    end
  end
end
```

To enable Amber tests with garnet just have to overwrite the Garnet
handler in spec helper

```crystal
handler = Amber::Server.instance.handler
handler.prepare_pipelines

module GarnetSpec
  HANDLER = handler
end
```

### CODE REVIEW

File: spec/controller/test_spec.cr [hunk 1]
```diff
@@ -1,77 +0,0 @@
...
-      headers.not_nil!.["Content-Type"].should eq ["application/x-www-form-urlencoded"]
-    end
-
-    it "should not force override of Content-Type" do
-      raw_response = subject.post("/posts", headers: HTTP::Headers{"Content-Type" => "application/json"})
-      response = JSON.parse(raw_response.body)
-      headers = response["headers"]?
-      headers.should_not be_nil
-      headers.not_nil!.["Content-Type"].should eq ["application/json"]
-    end
```
Comment (katafrakt) [inline spec/controller/test_spec.cr:57]: Those specs are gone after changes. Is it possible to bring them back?

Comment (eliasjpr) [reply spec/controller/test_spec.cr:57]: Yes you’re right :)

Comment (eliasjpr) [reply spec/controller/test_spec.cr:57]: I have added these back @katafrakt

**Comment by eliasjpr** [comment]
@robacarp @drujensen @faustinoaq please review this PR and let me know if this will be merged. Thanks

File: spec/spec_helper.cr [hunk 2]
```diff
@@ -1,29 +1,13 @@
...
+  def call(context)
+    response = { headers: context.request.headers.to_h }.to_json
+    context.response.write(response.to_slice)
+  end
+end
 
-  # Not all server implementations will support every WebDriver feature.
-  # Therefore, the client and server should use JSON objects with the properties
-  # listed below when describing which features a session supports.
-  capabilities = {
```
Comment (eliasjpr) [inline spec/spec_helper.cr:11]: There is no need to provide these options in spec helper since there are defaulted anyways.

**Review by robacarp — APPROVED** [review]

**Comment by eliasjpr** [comment]
Ping! @amberframework/contributors

**Review by drujensen — APPROVED** [review]

**Pull request merged**